src/Stratis.Features.FederatedPeg/FederatedPegBlockDefinition.cs

```diff
@@ -46,11 +51,36 @@ public class FederatedPegBlockDefinition : SmartContractPoABlockDefinition
 
         public override BlockTemplate Build(ChainedHeader chainTip, Script scriptPubKey)
         {
-            Script rewardScript = (chainTip.Height + 1) == this.Network.Consensus.PremineHeight
-                                   ? this.payToMultisigScript
-                                   : this.payToMemberScript;
+            // TODO: Shouldn't this be mining to a local wallet. Use this variable ^^
+
+            bool miningPremine = (chainTip.Height + 1) == this.Network.Consensus.PremineHeight;
+
+            Script rewardScript = miningPremine ? this.payToMultisigScript : this.payToMemberScript;
+
+            BlockTemplate built = base.Build(chainTip, rewardScript);
+
+            if (miningPremine)
+            {
+                this.BreakUpPremineCoinbase();
+            }
+
+            return built;
+        }
+
+        private void BreakUpPremineCoinbase()
```

Comment:
I know this is a pretty simple method, but do you think there's any value into breaking it out so it can be tested?

Reply:
Have done so